src/tools/singlejar/output_jar.h

```diff
@@ -15,6 +15,9 @@
 #ifndef SRC_TOOLS_SINGLEJAR_COMBINED_JAR_H_
 #define SRC_TOOLS_SINGLEJAR_COMBINED_JAR_H_
 
+// Must be included before anything else.
+#include "src/tools/singlejar/port.h"
```

Comment:
This file doesn't need the whole `port.h`, just the `off64_t` type definition. Could you split that out so including `port.h` (and its transitive closure of other header files) is unnecessary here?

Reply:
> I need to include <sys/stat.h> for all platforms to get off_t and off64_t.

I suppose you meant that line as a reply to this comment of mine.
I think it's OK to include what's necessary to define `off64_t`, so `<sys/stat.h>` is fine, but not more. Including things like `<io.h>` on Windows and all the other system headers is too much -- they might pollute the preprocessor namespace and global namespace.